addons/mail/migrations/12.0.1.0/post-migration.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2018 Eficent <http://www.eficent.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+from psycopg2.extensions import AsIs
+
+
+def fill_mail_blacklist_res_partner(cr):
+    cr.execute(
+        """
+        INSERT INTO mail_blacklist (email, active,
+            create_uid, create_date, write_uid, write_date)
+        SELECT email, active, create_uid, create_date, write_uid, write_date
+        FROM res_partner rp
+        WHERE rp.%s
+            AND rp.email NOT IN (SELECT email FROM mail_blacklist)
```

Comment:
Instead of this, you can use `GROUP BY email` and put `MAX` group operator in the rest of fields.

Reply:
max(active)? active is boolean :S

Reply:
Right, for that field, you can use `BOOL_OR(active)`.

Reply:
but what happens if active = true is and old record and active = false is a newer record, both having same mail? if grouped, they will be put in the blacklist as active = true.

I think the good approach is not grouping, just ordering the records by id DESC before the insert

Reply:
The problem here is performance doing this, and I'm not sure about the correctness of this any way. The case you are talking about can be the same conflictive if you have disabled and blacklisted the last created contact and leave the other one. Take into account that having one archived contact blacklisted and the non blacklisted one active, is not going to disable the sending, as the blacklist record will be archived as well.

Thus, my recommendation is still to group with OR.

Reply:
Also MAX(create_uid) doesn't makes sense. It should have the last create_uid, not the max :S

Reply:
Not really. Use `MIN` or `MAX` as you prefer, but selecting the last or the first one doesn't follow any business logic.